packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -198,16 +201,19 @@ export class KubernetesClient {
     this.kubeConfigWatcher.onDidChange(async () => {
       this._onDidUpdateKubeconfig.fire({ type: 'UPDATE', location });
       await this.refresh();
+      this.apiSender.send('kubernetes-context-update');
```

Comment:
as reported before I think we would need to subscribe to the onDidUpdateKubeconfig to fire the event when it's there
so it avoids to add apiSender on kubeConfigWatcher as we've already a listener that can call our apiSender method

Reply:
No need to add another onDidUpdateKubeconfig as we do that in kube context: https://github.com/containers/podman-desktop/blob/d64d20c7c4b50b525e10280148364505da5dff9c/extensions/kube-context/src/extension.ts#L158 

When I added an onDidUpdateKubeconfig outside of the extension, it was firing twice / refreshing twice all the time. 

Since kube context is in an extension that can be disable / enabled, this makes sense.